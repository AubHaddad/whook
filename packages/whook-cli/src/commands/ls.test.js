import initLsCommand, { definition as initLsCommandDefinition } from './ls';
import initEnvCommand, { definition as initEnvCommandDefinition } from './env';
import YError from 'yerror';

describe('lsCommand', () => {
  const log = jest.fn();
  const readDir = jest.fn();
  const _require = jest.fn();

  beforeEach(() => {
    log.mockReset();
    readDir.mockReset();
    _require.mockReset();
  });

  describe('should work', () => {
    it('with no plugin', async () => {
      const lsCommand = await initLsCommand({
        WHOOK_PLUGINS: [],
        WHOOK_PLUGINS_PATHS: [],
        readDir,
        log,
        EOL: '\n',
        require: _require,
        args: {
          _: ['ls'],
        },
      });
      const result = await lsCommand();

      expect({
        result,
        logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
        readDirCalls: readDir.mock.calls,
        requireCalls: _require.mock.calls,
      }).toMatchSnapshot();
    });

    it('with some plugins', async () => {
      readDir.mockResolvedValueOnce(['ls', 'env']);
      readDir.mockRejectedValueOnce(new YError('E_NO_MODULE'));
      _require.mockReturnValueOnce({
        default: initLsCommand,
        definition: initLsCommandDefinition,
      });
      _require.mockReturnValueOnce({
        default: initEnvCommand,
        definition: initEnvCommandDefinition,
      });

      const lsCommand = await initLsCommand({
        WHOOK_PLUGINS: ['@whook/cli', '@whook/whook'],
        WHOOK_PLUGINS_PATHS: [
          '/var/lib/node/node_modules/@whook/cli/dist',
          '/var/lib/node/node_modules/@whook/lol/dist',
        ],
        readDir,
        log,
        EOL: '\n',
        require: _require,
        args: {
          _: ['ls'],
        },
      });
      await lsCommand();

      expect({
        logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
        readDirCalls: readDir.mock.calls,
        requireCalls: _require.mock.calls,
      }).toMatchSnapshot();
    });

    it('with some plugins and a verbose output', async () => {
      readDir.mockResolvedValueOnce(['ls', 'env']);
      readDir.mockRejectedValueOnce(new YError('E_NO_MODULE'));
      _require.mockReturnValueOnce({
        default: initLsCommand,
        definition: initLsCommandDefinition,
      });
      _require.mockReturnValueOnce({
        default: initEnvCommand,
        definition: initEnvCommandDefinition,
      });

      const lsCommand = await initLsCommand({
        WHOOK_PLUGINS: ['@whook/cli', '@whook/whook'],
        WHOOK_PLUGINS_PATHS: [
          '/var/lib/node/node_modules/@whook/cli/dist',
          '/var/lib/node/node_modules/@whook/lol/dist',
        ],
        readDir,
        log,
        EOL: '\n',
        require: _require,
        args: {
          _: ['ls'],
          verbose: true,
        },
      });
      await lsCommand();

      expect({
        logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
        readDirCalls: readDir.mock.calls,
        requireCalls: _require.mock.calls,
      }).toMatchSnapshot();
    });
  });
});
