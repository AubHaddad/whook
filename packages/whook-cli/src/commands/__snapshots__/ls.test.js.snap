// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lsCommand should work with no plugin 1`] = `
Object {
  "logCalls": Array [],
  "readDirCalls": Array [],
  "requireCalls": Array [],
  "result": undefined,
}
`;

exports[`lsCommand should work with some plugins 1`] = `
Object {
  "logCalls": Array [
    Array [
      "debug",
      "✅ - No commands folder found at path \${pluginPath}",
    ],
    Array [
      "info",
      "

# Provided by \\"@whook/cli\\": 2 commands",
    ],
    Array [
      "info",
      "- ls: Print available commands",
    ],
    Array [
      "info",
      "- env: A command printing env values",
    ],
    Array [
      "info",
      "

# Provided by \\"@whook/whook\\": none",
    ],
  ],
  "readDirCalls": Array [
    Array [
      "/var/lib/node/node_modules/@whook/cli/dist/commands",
    ],
    Array [
      "/var/lib/node/node_modules/@whook/lol/dist/commands",
    ],
  ],
  "requireCalls": Array [
    Array [
      "/var/lib/node/node_modules/@whook/cli/dist/commands/ls",
    ],
    Array [
      "/var/lib/node/node_modules/@whook/cli/dist/commands/env",
    ],
  ],
}
`;

exports[`lsCommand should work with some plugins and a verbose output 1`] = `
Object {
  "logCalls": Array [
    Array [
      "debug",
      "✅ - No commands folder found at path \${pluginPath}",
    ],
    Array [
      "info",
      "

# Provided by \\"@whook/cli\\": 2 commands
",
    ],
    Array [
      "info",
      "- ls: Print available commands
$ whook ls
",
    ],
    Array [
      "info",
      "- env: A command printing env values
$ whook env --name NODE_ENV --default \\"default value\\"
",
    ],
    Array [
      "info",
      "

# Provided by \\"@whook/whook\\": none
",
    ],
  ],
  "readDirCalls": Array [
    Array [
      "/var/lib/node/node_modules/@whook/cli/dist/commands",
    ],
    Array [
      "/var/lib/node/node_modules/@whook/lol/dist/commands",
    ],
  ],
  "requireCalls": Array [
    Array [
      "/var/lib/node/node_modules/@whook/cli/dist/commands/ls",
    ],
    Array [
      "/var/lib/node/node_modules/@whook/cli/dist/commands/env",
    ],
  ],
}
`;
